### PR TITLE
Add congressional districts

### LINKED ISSUE #51: Import shapefiles for congressional districts
Making notes here for now. Using the Tiger/Line® 2015 data from the US Census Bureau for the 114th Congress.

Using this blog post for reference: https://devmynd.com/blog/2013-7-postgis-rails-on-heroku/

Extracted the EPSG code from the `.prj` file:

```
GEOGCS["GCS_North_American_1983",DATUM["D_North_American_1983",SPHEROID["GRS_1980",6378137,298.257222101]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295]]
```

### PR DESCRIPTION
**Pull request by joshsmith**
This will close #51 and parts of #8.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -33,6 +36,8 @@ gem "paperclip"
 gem "aws-sdk"
 gem "sidekiq"
 
+gem "rgeo"
```
Comment (christopherstyles) [inline Gemfile]: This is already a dependency of `activerecord-postgis-adapter`, no need to include twice.

**Comment by joshsmith** [comment]
I'm still expecting to add a bit of documentation here and there.

One question I have concerns `CongressionalDistrict.for_location`: do we want to be able to return a single result, and if so, where? Right now this would return two districts if the point is along the boundary of those two districts.

File: spec/lib/movement_api/shapefile_importers/congressional_district_importer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        RGeo::Shapefile::Reader.open('spec/support/fixtures/shapefiles/congressional_districts/file.shp', srid: 4326) do |file|
+          expect(file.num_records).to eq 4
+        end
+      end
+
+      describe "#call" do
+        it "creates CongressionalDistrict models" do
+          expect { subject.call }.to change{CongressionalDistrict.count}.from(0).to(3)
+        end
+      end
```
Comment (begedin) [inline spec/lib/movement_api/shapefile_importers/congressional_district_importer_spec.rb:21]: I'm not sure what else we can spec for here? How difficult is it to create new shapefiles? The only thing I can think of regarding specs is to have multiple files with different content to account for geometry, CD114FP codes, etc.

If creating the files is too difficult, we could create mocks and pass them as a return value into `RGeo::Shapefile::Reader.open`

**Comment by begedin** [comment]
Just some info on how I had to setup my environment to get specs to pass
- `brew install postgis` -takes a while
- `bundle install`
- in case rspec has a test failing with a message that GEOS is not supported, also need to do the following
- `brew install geos`-> `which geos-config` should return a path here
- `brew install proj` -> `which proj` should return a path
- `gem install rgeo` so that rgeo is rebuilt properly

I believe the first two steps should be enough. My issue was that I ran the bundle before the brew command, so rgeo wasn't built with GEOS support included.

File: spec/lib/movement_api/shapefile_importers/congressional_district_importer_spec.rb (hunk 2 above)
Comment (joshsmith) [inline spec/lib/movement_api/shapefile_importers/congressional_district_importer_spec.rb:21]: Creating the files is really complex. Can you clarify about the mocks?

**Comment by joshsmith** [comment]
@begedin I also had to do the same due to GEOS. I wonder if there's something special we need to do on Heroku, too.

File: spec/lib/movement_api/shapefile_importers/congressional_district_importer_spec.rb (hunk 2 above)
Comment (begedin) [inline spec/lib/movement_api/shapefile_importers/congressional_district_importer_spec.rb:21]: I'm just thinking something like

``` Ruby

mock_shapefile = []

x.times do
  mock_attributes = {
    "STATEFP" => "50"
    "CD114FP" => "ZZ"
    "CDSESSN" => "something"
    ...
  } 
  mock_record = double("Something", attributes: mock_attributes)
  mock_shapefile.push(mock_record)
end

allow(RGeo::Shapefile::Reader).to receive(:open).and_return(mock_shapefile )
```

It's not as simple as I'd like it to be, but might be simpler than creating files for each combination.

**Comment by begedin** [comment]
My guess is, this order should work:

```
brew install geos
brew install proj
brew install postgis
bundle install
```

There should also be no redundance with that one.

The important part is, `geos` and `proj` need to be installed before the `rgeo` gem is installed (as part of `bundle install`). That way, the gem will be built properly.

**Pull request closed without merging**